### PR TITLE
207 authorization bugs

### LINKED ISSUE #207: Double-check authorization and workflow logic
- /applications/{id}/download-application/{lang} (`download_application`) doesn't perform authorization (e.g. that the lender is authorized to download the application). This most likely is a bug.
- /applications/submit (`update_apps_send_notifications`) doesn't check if the application is expired.

The following don't check the application's status:

- /applications/upload-document (`upload_document`)
- /applications/{id}/reject-application (`reject_application`)
- /applications/{id}/upload-compliance (`upload_compliance`)

The variable is named `OCP_cannot_modify` (now `OCP_CAN_MODIFY`), but it applies to the FIs (lenders) as well.

### PR DESCRIPTION
**Pull request by yolile**
Closes #207

### CODE REVIEW

File: app/routers/downloads.py [hunk 1]
```diff
@@ -78,6 +79,9 @@ async def download_application(
     :return: A streaming response with a zip file containing the documents.
     """
     with transaction_session(session):
+        dependencies.raise_if_unauthorized(
+            application, user, roles=(models.UserType.OCP, models.UserType.FI), scopes=(ApplicationScope.UNEXPIRED,)
+        )
```
Comment (jpmckinney) [inline app/routers/downloads.py:84]: Maybe add support for scopes to `get_scoped_publication_as_user`. You can then change `get_publication_as_user` to `get_scoped_publication_as_user`, with the same roles and scopes arguments.

File: app/routers/guest/applications.py [hunk 2]
```diff
@@ -464,7 +465,9 @@ async def update_apps_send_notifications(
     session: Session = Depends(get_db),
     client: CognitoClient = Depends(dependencies.get_cognito_client),
     application: models.Application = Depends(
-        dependencies.get_scoped_publication_as_guest_via_payload(statuses=(models.ApplicationStatus.ACCEPTED,))
+        dependencies.get_scoped_publication_as_guest_via_payload(
+            statuses=(models.ApplicationStatus.ACCEPTED,), scopes=(ApplicationScope.UNEXPIRED,)
```
Comment (jpmckinney) [inline app/routers/guest/applications.py:469]: I put statuses last for consistency.

```suggestion
            scopes=(ApplicationScope.UNEXPIRED,), statuses=(models.ApplicationStatus.ACCEPTED,)
```

**Review by jpmckinney — COMMENTED** [review]
Just a couple comments.

I'm not familiar enough with the workflow, so I assume the statuses removed from `USER_CAN_EDIT_AWARD_BORROWER_DATA` and the new guards to upload_compliance, download_application and update_apps_send_notifications are correct.

**Comment by yolile** [comment]
> the statuses removed from USER_CAN_EDIT_AWARD_BORROWER_DATA

Yes, in the past this same list was used for listing applications, for example, now it is only used for the editing endpoints so we only need the actual editable status.

File: app/routers/downloads.py (hunk 1 above)
Comment (yolile) [reply app/routers/downloads.py:84]: Sounds good. Now that I think about it, should the get_*publication_* methods should be named get_*application_* instead?

Comment (jpmckinney) [reply app/routers/downloads.py]: Uh, yes, haha, I must have been thinking about the registry subconsciously!

**Review by jpmckinney — APPROVED** [review]
Just double-check my merge conflict resolution: 8be926047f32bf84c5c3170d47bc7839c78a74ef

**Pull request merged**